src/views/CompanyPanelView/Components/CompaniesTable.jsx

```diff
@@ -60,6 +65,30 @@ export function CompaniesPanelTable () {
     })
   }
 
+  const getChangedCompanies = (companies, selected) => {
+    return {
+      companies: companies.filter(company => {
+        // eslint-disable-next-line camelcase
+        const { is_public, id } = company
+        // eslint-disable-next-line camelcase
+        if ((is_public && selected.includes(id)) || (!is_public && !selected.includes(id))) {
+          return false
+        }
+        return true
+        // eslint-disable-next-line camelcase
+      }).reduce((prev, { id, is_public }) => ({ ...prev, [id]: !is_public }), {})
+    }
+  }
+
+  const onSave = async (_) => {
```

Comment:
You'll need to add logic when users change public value, but they cancel.
The table shows those values ​​as if they had changed.